### DISCUSSION TITLE
Reference Manual

### DISCUSSION THREAD

**Discussion by BernardUK** [elan-language/LanguageAndIDE#943] [General]
The 4 December draft is looking very good. How would you like to receive comments on it? Some might be quite trivial, others less so.

**Comment by richardpawson** [comment]
1. Please make any comments/corrections on the most up to date version that you can. I made more (small) changes today.
2.  Corrections, or improved text, preferred over comments where possible - we have a lot of work on our plates right now, and would look positively on anything that reduces what remains to be done rather than increases it ;-) I am sure you understand what I am saying. 
3. Please make them directly in Word, with track changes on.

**Reply by BernardUK** [reply]
Word? I can see only the PDF. Here are my immediate noticings:
pp.10-12,15,35,39,43,59,63,67,74-75 Many points at which an automatic cross-reference hasn't worked ("Error! ...")
p.10  Identifer -> Identifier
p.20 Clead -> Clear
p.21 this this -> this
p.63 merh -> ?

**Reply by BernardUK** [reply]
Found a few more single letter corrections (identifer, iiterable) and made them experimentally in a downloaded copy of the Beta 5 docx file with tracking turned on. Presumably I should instead make a pull request for the file using git in the usual way?

**Reply by richardpawson** [reply]
Yes, please.

**Reply by BernardUK** [reply]
'In the usual way'! -- I haven't yet managed to create a local repo into which I can pull the file reference-manual.docx. Any tips?

**Reply by BernardUK** [reply]
I think forget this for now, as the Ref Man is under your vital attention. I'll just post here things like:
Reference Manual Beta 5 13-Dec-2024
p.94 Iiterable -> iterable

**Reply by richardpawson** [reply]
Thanks - updated.

**Reply by BernardUK** [reply]
Reference Manual Beta 5 13-Dec-2024 (under constant revision I think).
I have a growing list of (mostly) quite small corrections, but am wary of listing so many here, suspect that such subediting is premature, and am certain that Richard has more important calls on his time. I will keep logging them anyway until they can be applied efficiently.

**Reply by richardpawson** [reply]
Thanks Bernard. FYI I have just released Beta 5 (at https://elan-lang.org/beta as usual). It does have some breaking changes in the language and library - now listed at the front of the manual. If you have written any programs already they might not load. If you email me I will tell you about a trick to migrate them - but it is not something I want to make public! (rpawson @ nakedobjects dot org ).

I hope that is the last of the breaking changes - but 'famous last words' and all that!

**Comment by richardpawson** [comment]
Thanks. For future reference the Word (source) document - reference-manual.docx - can be found here:

https://github.com/elan-language/LanguageAndIDE/tree/main

I will correct the typos. Unfortunately, Word's own mechanism for saving as .pdf all-too-frequently mangles the cross-references. Some of them might be stale (i.e. my fault) but many are simply mangled by Word. I might at some future point abandon the conversion to .pdf altogether and just publish a link to the Word doc.

**Comment by BernardUK** [comment]
This discussion has been completely superseded by Html documentation methods plus changes to the language and to the team, so can be safely closed.
